game/database.cc

```diff
@@ -127,19 +127,30 @@ void Database::queryPerPlayerHiscore(std::ostream & os, std::string const& track
 	}
 }
 
-bool Database::hasHiscore(Song const& s) const 
+bool Database::hasHiscore(Song const& s) const
 try {
 	return m_hiscores.hasHiscore(m_songs.lookup(s).value());
 } catch (const std::exception &e) {
 	std::clog << "database/error: Invalid song ID for song: " + s.artist + " - " + s.title << std::endl;
 	return false;
 }
 
-unsigned Database::getHiscore(const Song& s) const try {
+unsigned Database::getHiscore(Song const& s) const try {
 	const auto songid = m_songs.lookup(s);
 	return m_hiscores.getHiscore(songid.value());
 } catch (const std::exception& e) {
 	std::clog << "database/error: Invalid song ID for song: " + s.artist + " - " + s.title << std::endl;
 	return 0;
 }
 
+unsigned Database::getHiscore(SongPtr const& s) const {
+	try {
+		auto const songid = m_songs.getSongId(s);
+
+		return m_hiscores.getHiscore(songid);
+	} catch (const std::exception& e) {
+		std::clog << "database/error: Invalid song ID for song: " + s->artist + " - " + s->title << std::endl;
+		return 0;
```

Comment:
I dislike the magic 0; either return an optional or let the exception go out. The caller need to know there is no highscore in some way

Reply:
:+1: